htlcswitch/switch.go

```diff
@@ -1663,6 +1663,31 @@ func (s *Switch) Start() error {
 	}
 	s.blockEpochStream = blockEpochStream
 
+	// Before starting the HTLC forwarder, we sync our pending payment
+	// store with the circuit map. This must be done to ensure the store
+	// won't contain pending payments that was never forwarded on the
+	// network. By deleting those pending payments that are not found in
+	// the circuit map, we ensure that the router won't wait for a result
+	// to be returned for this payment ID.
+	// NOTE: this assumes that the circuit map has already been trimmed,
```

Comment:
I don't think this last assumption is sound with the current implementation. IIRC, the circuit map as is, can contain circuits for payments they have either already been completed or failed on the network. 

cc @cfromknecht